lib-js/compiler.ts

```diff
@@ -43,10 +48,14 @@ export function executableCompiler(
       }
       return { stdout, stderr, status }
     },
+    shutdown() {
```

Comment:
Thanks for helping with this.

Instead of shimming the interface, consider making `Compiler` an abstract class with a default noop `shutdown()`, and then turning the function `executableCompiler()` into an `ExecutableCompiler` class that extends Compiler. Something like—

```
export abstract class Compiler {
  abstract compile(path: string, args: string[]): Promise<Stdio>;
  shutdown(): void {}
}

```

Reply:
done. but this actually results in more code, not in less.

Reply:
IMO sometimes more code helps readability